### PR TITLE
[Project] add Vampire dataset project in dev-1.x

### PR DESCRIPTION
**Pull request by Provable0816**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects, like MMDet or MMDet3D.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: projects/medical/2d_image/fluorescein_angriogram/vampire/README.md [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+All the commands below rely on the correct configuration of `PYTHONPATH`, which should point to the project's directory so that Python can locate the module files. In `vampire/` root directory, run the following line to add the current directory to `PYTHONPATH`:
+
+```shell
+export PYTHONPATH=`pwd`:$PYTHONPATH
+```
+
+### Dataset preparing
+
+- download dataset from [here](https://vampire.computing.dundee.ac.uk/vesselseg.html) and decompression data to path `'data/'`.
+- run script `"python tools/prepare_dataset.py"` to split dataset and change folder structure as below.
```
Comment (xiexinch) [inline projects/medical/2d_image/fluorescein_angriogram/vampire/README.md:76]: We can't find this script.

**Review by xiexinch — CHANGES_REQUESTED** [review]

File: projects/medical/2d_image/fluorescein_angriogram/vampire/README.md [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+  │   │   │   │   │   │   ├── images
+  │   │   │   │   │   │   │   ├── train
+  │   │   │   │   |   │   │   │   ├── xxx.png
+  │   │   │   │   |   │   │   │   ├── ...
+  │   │   │   │   |   │   │   │   └── xxx.png
+  │   │   │   │   │   │   ├── masks
+  │   │   │   │   │   │   │   ├── train
+  │   │   │   │   |   │   │   │   ├── xxx.png
+  │   │   │   │   |   │   │   │   ├── ...
+  │   │   │   │   |   │   │   │   └── xxx.png
```
Comment (xiexinch) [inline projects/medical/2d_image/fluorescein_angriogram/vampire/README.md:100]: Some training images have a `.bmp` suffix, will these images not be used?

Comment (Provable0816) [reply projects/medical/2d_image/fluorescein_angriogram/vampire/README.md:102]: We transformed the suffix to `.png`. The processing codes have been pushed and the readme file has been fixed.

File: projects/medical/2d_image/fluorescein_angriogram/vampire/tools/prepare_dataset.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
+import os
+import shutil
+
+from PIL import Image
+
+path = 'path/to/vampire'
+
+if not os.path.exists(os.path.join(path, 'images', 'train')):
+    os.mkdirs(os.path.join(path, 'images', 'train'))
```
Comment (xiexinch) [inline projects/medical/2d_image/fluorescein_angriogram/vampire/tools/prepare_dataset.py:9]:
```suggestion
    os.system(f'mkdir -p {os.path.join(path, "images", "train")}')
```

File: projects/medical/2d_image/fluorescein_angriogram/vampire/tools/prepare_dataset.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+if not os.path.exists(os.path.join(path, 'masks', 'train')):
+    os.mkdirs(os.path.join(path, 'masks', 'train'))
```
Comment (xiexinch) [inline projects/medical/2d_image/fluorescein_angriogram/vampire/tools/prepare_dataset.py:12]:
```suggestion
    os.system(f'mkdir -p {os.path.join(path, "masks", "train")}')
```

File: projects/medical/2d_image/fluorescein_angriogram/vampire/tools/prepare_dataset.py [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+imgs_amd14 = os.listdir(os.path.join(origin_data_path, 'AMD14'))
+imgs_ger7 = os.listdir(os.path.join(origin_data_path, 'GER7'))
+
+for img in imgs_amd14:
+    shutil.copy(
+        os.path.join(origin_data_path, 'AMD14', img),
+        os.path.join(path, 'images', 'train', img))
+    shutil.copy(
+        os.path.join(origin_data_path, 'AMD14-GT', img),
+        os.path.join(path, 'masks', 'train', img))
```
Comment (xiexinch) [inline projects/medical/2d_image/fluorescein_angriogram/vampire/tools/prepare_dataset.py:25]:
```suggestion
    # copy GT
    img_gt = img.replace('.png', '-GT.png')
    shutil.copy(
        os.path.join(origin_data_path, 'AMD14-GT', f'{img_gt}'),
        os.path.join(path, 'masks', 'train', img))
```

File: projects/medical/2d_image/fluorescein_angriogram/vampire/tools/prepare_dataset.py [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+for img in imgs_ger7:
+    shutil.copy(
+        os.path.join(origin_data_path, 'GER7', img),
+        os.path.join(path, 'images', 'train', img))
+    shutil.copy(
+        os.path.join(origin_data_path, 'GER7-GT', img),
+        os.path.join(path, 'masks', 'train', img))
```
Comment (xiexinch) [inline projects/medical/2d_image/fluorescein_angriogram/vampire/tools/prepare_dataset.py:33]:
```suggestion
    # copy GT
    img_gt = img.replace('.bmp', '-GT.png')
    img = img.replace('bmp', 'png')
    shutil.copy(
        os.path.join(origin_data_path, 'GER7-GT', img_gt),
        os.path.join(path, 'masks', 'train', img))
```

File: projects/medical/2d_image/fluorescein_angriogram/vampire/tools/prepare_dataset.py [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+
+imgs = os.listdir(os.path.join(path, 'images', 'train'))
+for img in imgs:
+    if not img.endswith('.png'):
+        im = Image.open(os.path.join(origin_data_path, 'images', 'train', img))
```
Comment (xiexinch) [inline projects/medical/2d_image/fluorescein_angriogram/vampire/tools/prepare_dataset.py:38]:
```suggestion
        im = Image.open(os.path.join(path, 'images', 'train', img))
```

File: projects/medical/2d_image/fluorescein_angriogram/vampire/configs/vampire_512x512.py [hunk 8]
```diff
@@ -0,0 +1,42 @@
+dataset_type = 'VampireDataset'
+data_root = 'data/vampire'
```
Comment (xiexinch) [inline projects/medical/2d_image/fluorescein_angriogram/vampire/configs/vampire_512x512.py:2]:
```suggestion
data_root = 'data'
```

**Review by xiexinch — APPROVED** [review]

**Pull request merged**